### PR TITLE
Add advisory for rgb

### PR DESCRIPTION
**Pull request by Shnatsel**

### CODE REVIEW

**Comment by Shnatsel** [comment]
@HeroicKatora I'd appreciate your input on this. Notably I'm filing this as an informational advisory and not an exploit.

**Comment by Shnatsel** [comment]
I consider 0.5.4 the first one affected because the relevant methods seem to be introduced in this commit: https://github.com/kornelski/rust-rgb/commit/d11ef3989e727f8c73440f38051b9388540f5ad0

File: crates/rgb/RUSTSEC-0000-0000.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+date = "2020-10-14"
+title = "Allows viewing and modifying arbitrary structs as bytes"
+url = "https://github.com/kornelski/rust-rgb/issues/35"
+informational = "unsound"
+keywords = ["type confusion"]
+description = """
+Affected versions of rgb crate allow viewing and modifying data any type `T` wrapped in `RGB<T>` as bytes,
+and do not correctly constrain `RGB<T>` and other wrapper structures to the types for which it is safe to do so.
+
+If a type containing madding is wrapped in `RGB<T>` and similar wrapper structures,
```
Comment (HeroicKatora) [inline crates/rgb/RUSTSEC-0000-0000.toml:13]:
```suggestion
If a type containing padding is wrapped in `RGB<T>` and similar wrapper structures,
```

File: crates/rgb/RUSTSEC-0000-0000.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+viewing it as bytes may lead to exposure of contents of uninitialized memory.
+
+If a type containing a pointer is wrapped in `RGB<T>` and similar wrapper structures,
+modifying it as bytes may lead to derefericing of arbitrary pointers.
```
Comment (HeroicKatora) [inline crates/rgb/RUSTSEC-0000-0000.toml:17]:
```suggestion
modifying it as bytes may lead to dereferencing of arbitrary pointers.
```

File: crates/rgb/RUSTSEC-0000-0000.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[advisory]
+id = "RUSTSEC-0000-0000"
+package = "rgb"
+date = "2020-10-14"
+title = "Allows viewing and modifying arbitrary structs as bytes"
+url = "https://github.com/kornelski/rust-rgb/issues/35"
+informational = "unsound"
+keywords = ["type confusion"]
+description = """
+Affected versions of rgb crate allow viewing and modifying data any type `T` wrapped in `RGB<T>` as bytes,
```
Comment (HeroicKatora) [inline crates/rgb/RUSTSEC-0000-0000.toml:10]:
```suggestion
Affected versions of rgb crate allow viewing and modifying data of any type `T` wrapped in `RGB<T>` as bytes,
```

**Comment by Shnatsel** [comment]
Better now?

File: crates/rgb/RUSTSEC-0000-0000.toml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[advisory]
+id = "RUSTSEC-0000-0000"
+package = "rgb"
+date = "2020-10-14"
```
Comment (lzutao) [inline crates/rgb/RUSTSEC-0000-0000.toml:4]: in the future?

Comment (Shnatsel) [reply crates/rgb/RUSTSEC-0000-0000.toml:4]: Gah, my brain must have melted from all the heat. Sorry. Will fix that now.

Comment (Shnatsel) [reply crates/rgb/RUSTSEC-0000-0000.toml:4]: Fixed in git master. Thanks for spotting!

**Pull request merged**